library/manageiq.py

```diff
@@ -91,7 +110,20 @@ def update_required(self, provider_id, hostname, port):
         except Exception as e:
             self.module.fail_json(msg="Failed to get provider data. Error: %s" % e)
         endpoint = result['endpoints'][0]
-        return False if (endpoint['hostname'] == hostname and endpoint['port'] == int(port)) else True
+
+        if (endpoint['hostname'] == endpoints[0]['endpoint']['hostname'] and
+            endpoint['port'] == int(endpoints[0]['endpoint']['port'])):
+            if self.module.params['metrics']:
+                h_endpoint = result['endpoints'][1]
+                if (h_endpoint['hostname'] == endpoints[1]['endpoint']['hostname'] and
+                    h_endpoint['port'] == int(endpoints[1]['endpoint']['port'])):
+                    return False
+                else:
+                    return True
+            else:
+                return False
+        else:
+            return True
```

Comment:
@dkorn it seems that you should change the logic in order to have less nested if's.